src/map/channel.hpp

```diff
@@ -16,21 +19,20 @@ struct DBMap;
 #define CHAN_NAME_LENGTH 20
 #define CHAN_MSG_LENGTH 150
 
-enum Channel_Opt {
-	CHAN_OPT_NONE		    = 0,	///< None
-	CHAN_OPT_ANNOUNCE_SELF  = 0x01,	///< Shows info when player joined/left channel to self
-	CHAN_OPT_ANNOUNCE_JOIN  = 0x02,	///< Shows info if player joined the channel
-	CHAN_OPT_ANNOUNCE_LEAVE = 0x04,	///< Shows info if player left the channel
-	CHAN_OPT_MSG_DELAY	    = 0x08,	///< Enables chat delay
-	CHAN_OPT_COLOR_OVERRIDE = 0x10,	///< Enables color channel be override by player's font color
-	CHAN_OPT_CAN_CHAT		= 0x20,	///< Allows player to chat in the channel
-	CHAN_OPT_CAN_LEAVE		= 0x40,	///< Allows player to leave the channel
-	CHAN_OPT_AUTOJOIN		= 0x80,	///< Player will be autojoined to the channel
-
-	CHAN_OPT_BASE = CHAN_OPT_ANNOUNCE_SELF|CHAN_OPT_MSG_DELAY|CHAN_OPT_CAN_CHAT|CHAN_OPT_CAN_LEAVE,
+enum e_channel_options : uint16 {
```

Comment:
Should we utilize some C++ features like std::bitset? We've used it several times before to shy away from masking. I guess it's not super major in this case, but for the skill/status databases for example, the enum rose over int32 because of all the options and at the time the script engine wouldn't parse above that.

Reply:
We can in the next pr, this was just to remove the libconfig dependency